### ISSUE TITLE
empty collection creates query selecting all datasets?

### ISSUE THREAD

**Issue by JJ-Author** [dbpedia/databus#16]
the default query for an empty collection can be very expensive. one click to preview will result in a query that returns all datasets,
I suggest to not create a query or disable preview if no data is added

**Comment by holycrab13** [comment]
Needs some discussion on how to do this:

```
PREFIX dcv: <http://dataid.dbpedia.org/ns/cv#>
PREFIX dct:    <http://purl.org/dc/terms/>
PREFIX dcat:   <http://www.w3.org/ns/dcat#>
PREFIX rdf:    <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs:   <http://www.w3.org/2000/01/rdf-schema#>
SELECT ?file WHERE
{
	{
		GRAPH ?g
		{
			?dataset dcat:distribution ?distribution .
			?distribution dataid:file ?file .
			?distribution a dataid:Nonsense .
		}
	}
	UNION
	{
		SERVICE <https://dev.databus.dbpedia.org/sparql>
		{
			GRAPH ?g
			{
				?dataset dcat:distribution ?distribution .
				?distribution dataid:file ?file .
				?distribution a dataid:Nonsense .
			}
		}
	}
}
```

The above works but is ugly - it appends some random constraint to a source node (Databus node) with no child nodes.
I could also add a LIMIT 0 to the whole query if ANY of the Databus nodes has no children

**Comment by holycrab13** [comment]
FILTER is probably the 'cleanest' way to do this - but then the query is still expensive.

**Comment by holycrab13** [comment]
should be resolved, reopen if persists in 2.1.0
